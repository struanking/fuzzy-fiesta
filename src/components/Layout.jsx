import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

const Header = styled.header`
  margin: 2rem auto 3rem;
  width: 100%;
`;

const SiteTitle = styled.h1`
  color: #385b66;
  font-size: 3rem;
  font-weight: normal;
  margin: 0;
  text-align: center;
`;

const Layout = ({ children }) => (
  <div>
    <Header role="banner">
      <SiteTitle>Energy Usage and Meter Readings</SiteTitle>
    </Header>

    <main id="main">
      {children}
    </main>
  </div>
);

Layout.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element),
  ]).isRequired,
};

export default Layout;
