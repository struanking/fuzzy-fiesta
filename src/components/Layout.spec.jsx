import React from 'react';
import renderer from 'react-test-renderer';
import Layout from './Layout';

test('renders Layout correctly', () => {
  const tree = renderer
    .create(<Layout><p>content</p></Layout>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
