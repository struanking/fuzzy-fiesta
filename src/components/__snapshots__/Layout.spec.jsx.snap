// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Layout correctly 1`] = `
Array [
  <header
    className="sc-bdVaJa fVtXgW"
    role="banner"
  >
    <h1
      className="sc-bwzfXH gusAcy"
    >
      Energy Usage and Meter Readings
    </h1>
  </header>,
  <main
    id="main"
  >
    <p>
      content
    </p>
  </main>,
  <footer
    className="sc-htpNat cdOTFi"
    role="contentinfo"
  >
    <p>
      Struan King | Bulb Energy | 2018
    </p>
  </footer>,
]
`;
